scene/3d/mesh_instance.cpp

```diff
@@ -821,6 +821,323 @@ void MeshInstance::create_debug_tangents() {
 	}
 }
 
+bool MeshInstance::is_mergeable_with(const MeshInstance &p_other) {
+	if (!get_mesh().is_valid() || !p_other.get_mesh().is_valid()) {
+		return false;
+	}
+
+	Ref<Mesh> rmesh_a = get_mesh();
+	Ref<Mesh> rmesh_b = p_other.get_mesh();
+
+	int num_surfaces = rmesh_a->get_surface_count();
+	if (num_surfaces != rmesh_b->get_surface_count()) {
+		return false;
+	}
+
+	for (int n = 0; n < num_surfaces; n++) {
+		// materials must match
+		if (get_active_material(n) != p_other.get_active_material(n)) {
+			return false;
+		}
+
+		// formats must match
+		uint32_t format_a = rmesh_a->surface_get_format(n);
+		uint32_t format_b = rmesh_b->surface_get_format(n);
+
+		if (format_a != format_b) {
+			return false;
+		}
+	}
+
+	//	if (get_surface_material_count() != p_other.get_surface_material_count()) {
+	//		return false;
+	//	}
+
+	//	for (int n = 0; n < get_surface_material_count(); n++) {
+	//		if (get_surface_material(n) != p_other.get_surface_material(n)) {
+	//			return false;
+	//		}
+	//	}
+
+	// test only allow identical meshes
+	//	if (get_mesh() != p_other.get_mesh()) {
+	//		return false;
+	//	}
```

Comment:
Dead code?

Reply:
Yes, but in this case it is deliberate, this is in the mesh merging code, deciding the circumstances where meshes are close enough to merge. There is a scale of how conservative we can be for allowing merging, and I'm not absolutely sure ahead of time where this should be. So I figured we should start by being very liberal about allowing merging, then if we get bug reports with this feature, we can re-enable 1 or more of these sections.

Perhaps if I put a note to this effect it will be a good compromise. (We could alternatively remove the code entirely, but that might be a pain for anyone debugging this feature at a later date.)

I'm not absolutely sure on what this should be, because there are a lot of mesh configurations available in Godot and I'm unlikely to think of all of them in advance.

Incidentally, if this feature works out ok it will also be very good for use outside of rooms & portals, as a lot of users have been asking for mesh merging / static batching capabilities.

Reply:
Yeah I would just add an explanatory comment then, so that we know what's the purpose of keeping this code commented out.